cudamapper/src/overlapper.cpp

```diff
@@ -191,8 +191,9 @@ void Overlapper::print_paf(const std::vector<Overlap>& overlaps, const std::vect
                     overlap.target_length_,
                     overlap.target_start_position_in_read_,
                     overlap.target_end_position_in_read_,
-                    overlap.num_residues_,
-                    0,
+                    overlap.num_residues_ * k, // Print out the number of residue matches multiplied by kmer size to get approxuimate number of matching bases
+                    std::max(std::abs(int(overlap.target_start_position_in_read_) - int(overlap.target_end_position_in_read_)),
```

Comment:
What kind of safe cast do you mean? `uint32_t` -> `int32_t` which checks that the values is representable in `int32_t`? We can think about that, but I think it is not needed that often.

Can `overlap.target_start_position_in_read_` ever be larger than `overlap.target_end_position_in_read_`?

Reply:
> What kind of safe cast do you mean? uint32_t -> int32_t which checks that the values is representable in int32_t? We can think about that, but I think it is not needed that often.

yes, having something like that

> Can overlap.target_start_position_in_read_ ever be larger than overlap.target_end_position_in_read_?

I think so, in cases of inverse direction... @vellamike ?

Reply:
> Can overlap.target_start_position_in_read_ ever be larger than overlap.target_end_position_in_read_?

Yes, it is possible as Milos says. Even if it wasn't, I don't want the function to have that assumption.

The alternative is to check which one is bigger and do the subtraction in the correct order but this seems messy/expensive.

Reply:
I have changed this to use static_cast, but not sure if I should do anything else here.

Reply:
I think it would be good to have a function which does an `assert()` of unsigned number to make sure it fits the signed number, but as this is not the only place where something like that is missing let's leave it as it is